flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/rules/logical/PushProjectIntoTableSourceScanRule.scala

```diff
@@ -33,6 +32,9 @@ class PushProjectIntoTableSourceScanRule extends RelOptRule(
   override def matches(call: RelOptRuleCall): Boolean = {
     val scan: FlinkLogicalTableSourceScan = call.rel(1).asInstanceOf[FlinkLogicalTableSourceScan]
     scan.tableSource match {
+      // projection pushdown is not supported for sources that provide time indicators
+      case r: DefinedRowtimeAttribute if r.getRowtimeAttribute != null => false
```

Comment:
Can you create a follow-up issue for this?

Reply:
Yes will do, but it is not easy to solve IMO.